### PR TITLE
Updated README with details for Config options

### PR DESCRIPTION
**Pull request by alfetopito**
Depends on #987 

~Pending legal review regarding mention to DxDao's TCR and wording.~ Done

Superseding https://github.com/gnosis/dex-react/pull/997

### CODE REVIEW

**Comment by gnosis-info** [comment]
Travis automatic deployment:
 https://pr1005--dexreact.review.gnosisdev.com

File: README.md [hunk 1]
```diff
@@ -111,7 +111,31 @@ import About from 'pages/About'
 
 ## Config
 
-### TCR (Optional)
+Default app configs can be found on [the default config file](./config-default.yaml)
+
+We recommend against editing this file directly, though.
+
+If you wish to replace any default config, create a file named `config` inside the [`custom` folder](./custom).
```
Comment (anxolin) [inline README.md:118]: Probably we can create already the custom folder and leave it empty, just leave an empty `.gitkeep` file

File: README.md [hunk 2]
```diff
@@ -111,7 +111,31 @@ import About from 'pages/About'
...
+
+Both JSON and YAML formats are supported.
+
+Simply replace any config found on [config-default](./config-default.yaml).
+
+Below we provide details for each config.
+
+### `name`
```
Comment (anxolin) [inline README.md:126]: Maybe we can use a table like markdown? showing the name, the description, the default value

Comment (anxolin) [reply README.md:126]: I continued reading, and a table maybe makes it more complicated for the TCR part, cause it's a bit complex. So maybe just ignore my comment if there's no easy way to make a compact description of the parameters

Ultimately, I think we might find ourselves replicating between the README.md and the `config-default.yaml` so it's not bad to consider being brief in the readme, and use comments in the default file to document things if we come to that point.

File: README.md (hunk 1 above)
Comment (alfetopito) [reply README.md:118]: I did that at some point but I guess it was lost in a merge/rebase. Will add it.

File: README.md (hunk 2 above)
Comment (alfetopito) [reply README.md:126]: We can split this into a separated readme/wiki/etc for config if this becomes to big.
For me this is still manageable. I've seem much larger readmes out there :)

Comment (anxolin) [reply README.md:126]: Cool!

File: webpack.config.babel.js [hunk 3]
```diff
@@ -155,25 +158,24 @@ module.exports = ({ stats = false } = {}) => ({
...
-      'process.env.MOCK_WALLET': JSON.stringify(process.env.MOCK_WALLET || process.env.MOCK || 'false'),
-      'process.env.MOCK_TOKEN_LIST': JSON.stringify(process.env.MOCK_TOKEN_LIST || process.env.MOCK || 'false'),
-      'process.env.MOCK_ERC20': JSON.stringify(process.env.MOCK_ERC20 || process.env.MOCK || 'false'),
-      'process.env.MOCK_DEPOSIT': JSON.stringify(process.env.MOCK_MOCK_DEPOSIT || process.env.MOCK || 'false'),
-      'process.env.MOCK_EXCHANGE': JSON.stringify(process.env.MOCK_EXCHANGE || process.env.MOCK || 'false'),
-      'process.env.MOCK_WEB3': JSON.stringify(process.env.MOCK_WEB3 || process.env.MOCK || 'false'),
-
-      // AUTOCONNECT: only applies for mock implementation
-      'process.env.AUTOCONNECT': JSON.stringify(process.env.AUTOCONNECT || 'true'),
+      CONFIG: JSON.stringify(config),
```
Comment (Velenir) [inline webpack.config.babel.js:178]: I still think this is a bad idea on account of no treeshaking
I still think a good loader would be better
I'll need experiment with that

Comment (alfetopito) [reply webpack.config.babel.js:178]: You mean putting all the config into a global var?
In terms of the effects on treeshaking, how big does the config have to be in order to have an actual impact? I have no idea.

Comment (Velenir) [reply webpack.config.babel.js:178]: I'll test stuff tomorrow. Will see

Comment (alfetopito) [reply webpack.config.babel.js]: :crossed_fingers:

**Pull request merged**